cmd/kubeadm/app/preflight/checks.go

```diff
@@ -514,7 +516,18 @@ func (kubever KubeletVersionCheck) Check() (warnings, errors []error) {
 	if kubeletVersion.LessThan(kubeadmconstants.MinimumKubeletVersion) {
 		return nil, []error{fmt.Errorf("Kubelet version %q is lower than kubadm can support. Please upgrade kubelet", kubeletVersion)}
 	}
-	return nil, []error{}
+
+	if kubever.KubernetesVersion != "" {
+		k8sVersion, err := versionutil.ParseSemantic(kubever.KubernetesVersion)
+		if err != nil {
+			return nil, []error{fmt.Errorf("couldn't parse kubernetes version %q: %v", kubever.KubernetesVersion, err)}
+		}
+		firstUnsupportedKubeletVersion := versionutil.MustParseSemantic(fmt.Sprintf("%d.%d.%s", k8sVersion.Major(), k8sVersion.Minor()+1, "0-0"))
```

Comment:
This can be simpler I think... Why not just do `if kubeletVersion.Minor() > k8sVersion.Minor()` => error?

Reply:
While developing next minor version, it is fine to have kubelet a bit newer than control plane. 
e.g. testing kubelet 1.9.0-alpha.3+gitXYZ on top of 1.8.x stable cluster. Just be a bit more handy for developers.
Release 1.9.0 will complain, but development builds before 1.9.0 release will be ok.